cdap-client/src/main/java/co/cask/cdap/client/NamespaceClient.java

```diff
@@ -18,34 +18,21 @@
 
 import co.cask.cdap.client.config.ClientConfig;
 import co.cask.cdap.client.util.RESTClient;
-import co.cask.cdap.common.conf.Constants;
-import co.cask.cdap.common.exception.AlreadyExistsException;
-import co.cask.cdap.common.exception.BadRequestException;
-import co.cask.cdap.common.exception.CannotBeDeletedException;
-import co.cask.cdap.common.exception.NotFoundException;
 import co.cask.cdap.common.exception.UnAuthorizedAccessTokenException;
-import co.cask.cdap.proto.NamespaceMeta;
-import co.cask.common.http.HttpMethod;
+import co.cask.cdap.common.namespace.AbstractNamespaceClient;
 import co.cask.common.http.HttpRequest;
 import co.cask.common.http.HttpResponse;
-import co.cask.common.http.ObjectResponse;
-import com.google.common.reflect.TypeToken;
-import com.google.gson.Gson;
-import com.google.gson.JsonObject;
 
 import java.io.IOException;
 import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
 import java.net.URL;
-import java.util.List;
 import javax.inject.Inject;
 
 /**
  * Client to interact with CDAP namespaces
  */
-public class NamespaceClient {
-  private static final String NAMESPACE_ENTITY_TYPE = "namespace";
-  private static final Gson GSON = new Gson();
-
+public class NamespaceClient extends AbstractNamespaceClient {
```

Comment:
Since its in cdap-clients, I wanted its name to be consistent with other clients (i.e. DatasetClient, ApplicationClient, StreamClient)